module/zfs/abd.c

```diff
@@ -1374,7 +1374,8 @@ abd_raidz_gen_iterate(abd_t **cabds, abd_t *dabd,
 		 */
 		ASSERT3U(((uint64_t)len & 511ULL), ==, 0);
 
-		func_raidz_gen(caddrs, daiter.iter_mapaddr, len, dlen);
+		if (dabd && dsize > 0)
```

Comment:
I think this won't work. `func_raidz_gen()` is sometimes called only on code buffers.
You can just init daiter initially ti silence this warning:
```c
struct abd_iter daiter = { 0 };
```